responsemanager/queryexecutor/queryexecutor.go

```diff
@@ -123,30 +119,28 @@ func (qe *QueryExecutor) executeQuery(
 	// Execute the traversal operation, continue until we have reason to stop (error, pause, complete)
 	err := qe.runTraversal(p, rt)
 
+	_, isPaused := err.(hooks.ErrPaused)
+	if isPaused {
+		return err
+	}
+
+	if err == ErrNetworkError || ipldutil.IsContextCancelErr(err) {
+		rt.ResponseStream.ClearRequest()
+		return err
+	}
+
```

Comment:
What are the implications of moving these outside of a "transaction"? There's a similar change in responsemanager/server.go that moves the initial error & state checking outside of the `Transaction()` call. Are we losing anything by doing this?

Reply:
not really. The whole idea of Transaction is to make sure a series of operations go out in the same libp2p message over the wire (for example, a block + a pause message that got generated running hooks on the block). ClearRequest doesn't actually send anything over the wire. I have no idea why it was in transaction in the first place :)